repo/fsrepo/fsrepo.go

```diff
@@ -595,11 +596,43 @@ func (r *FSRepo) GetConfigKey(key string) (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
-	var cfg map[string]interface{}
-	if err := serialize.ReadConfigFile(filename, &cfg); err != nil {
+
+	var cfgMap map[string]interface{}
+	if err := serialize.ReadConfigFile(filename, &cfgMap); err != nil {
+		return nil, err
+	}
+
+	/*
+		Get an instance of Config for retrieving Config defined keys. Grabbing
+		values directly from Config allows us to retrieve values for keys that
+		may be flagged as 'json:",omitempty"' when the values of those keys are
+		considered empty.
+	*/
+	cfg, err := config.FromMap(cfgMap)
+	if err != nil {
 		return nil, err
 	}
-	return common.MapGetKV(cfg, key)
+
+	// Walk the Config instance looking for key fields
+	fieldValue := reflect.ValueOf(cfg)
+	parts := strings.Split(key, ".")
+	for i, part := range parts {
+		fieldValue = reflect.Indirect(fieldValue).FieldByName(part)
```

Comment:
This is making the sharness fail because we can't access maps like this:

Calling it with `Datastore.Spec.child.path`:
```
panic: reflect: call of reflect.Value.FieldByName on map Value

goroutine 1 [running]:
reflect.flag.mustBe(...)
	/usr/local/go/src/reflect/value.go:221
reflect.Value.FieldByName(0x1f10540, 0xc000c10478, 0x195, 0x7ffefcde6d28, 0x5, 0x1f10540, 0xc000c10478, 0x195)
	/usr/local/go/src/reflect/value.go:903 +0x25a
github.com/ipfs/go-ipfs/repo/fsrepo.(*FSRepo).GetConfigKey(0xc0002bbb00, 0x7ffefcde6d19, 0x19, 0x0, 0x0, 0x0, 0x0)
	src/github.com/ipfs/go-ipfs/repo/fsrepo/fsrepo.go:621 +0x387
```

Reply:
Solution direction changed, resolving conservation.